src/portfolio/portfolio.es6

```diff
@@ -0,0 +1,306 @@
+        row.data(cols);
+
+        /* colorize indicative column on change */
+        var tr = table.find('#' + id);
+        if(!contract.is_valid_to_sell) {
+            tr.removeClass('indicative-red indicative-green').addClass('resale-not-offered');
+        } else {
+            tr.removeClass('resale-not-offered');
+            if(perv_indicative !== bid_price) {
+            tr.removeClass('indicative-red indicative-green')
+                .addClass((perv_indicative*1 < bid_price*1) ? 'indicative-green' : 'indicative-red');
+            }
+        }
+    }
+}
+
+var subscribed_before = false;
+var subscribers = 0;
+liveapi.events.on('logout', function(){
+    subscribed_before = false;
+    subscribers = 0;
+});
+
+/* command could be 'subscribe','forget' and 'resubscribe'. */
+const proposalOpenContract = (command) => {
+    if(command === 'subscribe') {
+        ++subscribers;
+        if(!subscribed_before && subscribers > 0) {
+            liveapi.send({ proposal_open_contract: 1,subscribe: 1 })
+                .then(function(data){ subscribed_before = true; })
+                .catch(function (err) {
+                    console.error(err);
+                    $.growl.error({ message: err.message });
+                });
+        }
+    }
+    else if(command === 'forget') {
+        --subscribers;
+        if(subscribed_before && subscribers === 0) {
+            liveapi.send({ forget_all: 'proposal_open_contract' })
+                .then(function(data){
+                    subscribed_before = false;
+                })
+                .catch(function (err) {
+                    subscribed_before = false;
+                    console.error(err.message);
+                });
+        }
+    }
+    else if( command === 'resubscribe' ) {
+        liveapi.send({ forget_all: 'proposal_open_contract' })
+            .then(function(data) {
+                subscribed_before = false;
+                --subscribers;
+                proposalOpenContract('subscribe'); /* subscribe again */
+            })
+            .catch(function (err) {
+                subscribed_before = false;
+                console.error(err.message);
+            });
+    }
+    else {
+        console.error('wrong command!');
+        return;
+    }
+}
+
+var on_arrow_click = function(e){
+    var target = e.target;
+    var $target = $(target);
+    if(target.tagName !== 'BUTTON' || $target.hasClass('button-disabled'))
+    return;
+    var tr = target.parentElement.parentElement;
+    var transaction = table.api().row(tr).data();
+    transaction = _.last(transaction);
+    $target.addClass('button-disabled');
+    require(['viewtransaction/viewTransaction'], function(viewTransaction){
+    viewTransaction.init(transaction.contract_id, transaction.transaction_id)
+                    .then(function(){ $target.removeClass('button-disabled'); });
+    });
+}
+
+function initPortfolioWin() {
+    require(['css!portfolio/portfolio.css']);
```

Comment:
@aminroosta Since you have refactored this code and moved it to ES6, could you please do import statement instead of require. This way it would save one network call.